### PR TITLE
:zap: disable prefetch while with a slow network

### PR DESCRIPTION
**Pull request by kuitos**
This change is [Reviewable](https://reviewable.io/reviews/umijs/qiankun/211)

### CODE REVIEW

File: src/prefetch.ts [hunk 1]
```diff
@@ -25,15 +24,40 @@ declare global {
...
+        },
+      });
+    }, 1);
+  };
+
 /**
  * 预加载静态资源，不兼容 requestIdleCallback 的浏览器不做任何动作
  * @param entry
  * @param fetch
  */
```
Comment (howel52) [inline src/prefetch.ts]: setTimeout 模拟后，这个备注得改了

File: src/prefetch.ts [hunk 2]
```diff
@@ -25,15 +24,40 @@ declare global {
...
-export function prefetch(entry: Entry, fetch?: Fetch) {
-  const requestIdleCallback = window.requestIdleCallback || noop;
+export function prefetch(entry: Entry, fetch?: Fetch): void {
+  if (navigator.connection) {
+    // Don't prefetch if using 2G or if Save-Data is enabled.
+    if (navigator.connection.saveData || /2g/.test(navigator.connection.effectiveType)) return;
+  }
 
   requestIdleCallback(async () => {
```
Comment (howel52) [inline src/prefetch.ts]: requestIdleCallback(....) 要放在 else 里面么

Comment (kuitos) [reply src/prefetch.ts]: 不用，prettier 会自动格式化

File: src/prefetch.ts (hunk 1 above)
Comment (kuitos) [reply src/prefetch.ts]: done

**Pull request merged**